test/code/plugins/oms_common_test.rb

```diff
@@ -94,6 +100,256 @@ def teardown
       "OSAlias=UniversalR\n" \
       "OSManufacturer=Central Logistics GmbH\n"
 
+    # Begin:  Unit Tests for some simple methods.
+    # May best be put in another file later.
+
+    def test_clean_hostname_string
+      hn = ' www.five.com:333     '
```

Comment:
Can you comment on how we know that all of these test cases (particularly www.* and IP addresses) are confirmed possibilities when `Socket.gethostname()` returns? My main concern here is that we might be writing code for an unconfirmed issue. Do we have references (besides our "10" telemetry) showing that these are anomalies we have to account for?

Reply:
No.  I presumed this.  We may be able to trim out the concerns of FQDN.  Socket.gethostname should never return such a thing.  I likely went overboard here.  Rationale is as follows:

1. In my mind, I conceive both from experiences and knowledge of configurations many mis-configuration combinations of DNS and hostname.  This may just be my problem.
2.  At the very least, there is a clear justification for good logging on any situation which is out of the ordinary.  However, checking for www and FQDN inside a hostname clearly go beyond this need.
3.  Once I had the function programmed, I needed to write tests for it.  It is perfectly reasonable at this point to take the www check and the FQDN check out, as long as all odd cases are logged with warnings, I believe.

Reply:
I changed this a bit and took out literally all the FQDN stuff.  It's around, so we can bring it back later (and it was passing all the tests, too, but it still has risks, and strongly questionable need, so good for now).  Concise is good.

Reply:
"Socket.gethostname should never return such a thing" - this is what I wanted to know. If that is the case, then we should not work on checking for it.